src/models/index.js

```diff
@@ -24,6 +25,68 @@ if (config.use_env_variable) {
 
 audit.attachHooksForAuditing(sequelize);
 
+const descriptiveDetails = () => {
+  const loggedUser = httpContext.get('loggedUser') ? httpContext.get('loggedUser') : null;
+  const transactionId = httpContext.get('transactionId') ? httpContext.get('transactionId') : null;
+  const sessionSig = httpContext.get('sessionSig') ? httpContext.get('sessionSig') : null;
+  const impersonationId = httpContext.get('impersonationUserId') ? httpContext.get('impersonationUserId') : null;
+  const descriptor = httpContext.get('auditDescriptor') ? httpContext.get('auditDescriptor') : null;
+
+  return {
+    ...(descriptor && { descriptor }),
+    ...(loggedUser && { loggedUser }),
+    ...(impersonationId && { impersonationId }),
+    ...(sessionSig && { sessionSig }),
+    ...(transactionId && { transactionId }),
+  };
+};
+
+// Function to gracefully close the Sequelize connection
+const gracefulShutdown = async (msg) => {
+  const details = JSON.stringify(descriptiveDetails());
+  try {
+    await sequelize.close();
+    auditLogger.info(`Sequelize disconnected through ${msg}: ${details}`);
+  } catch (err) {
+    auditLogger.error(`Error during Sequelize disconnection through ${msg}: ${details}: ${err}`);
+  }
+};
+
+// Listen for SIGINT (Ctrl+C)
+process.on('SIGINT', async () => {
+  await gracefulShutdown('app termination (SIGINT)');
+  process.exit(0);
+});
+
+// Listen for SIGTERM (e.g., kill command)
+process.on('SIGTERM', async () => {
+  await gracefulShutdown('app termination (SIGTERM)');
+  process.exit(0);
+});
+
+// Listen for uncaught exceptions
+process.on('uncaughtException', async (err) => {
+  auditLogger.error('Uncaught Exception:', err);
+  await gracefulShutdown('uncaught exception');
+  process.exit(1);
+});
```

Comment:
We also watch for `uncaughtException` here:

https://github.com/HHS/Head-Start-TTADP/blob/main/src/app.js/#L25-L28

If the one in `src/app.js` happens first and the process is exited then and there, does THIS callback ever get called?

In the test you have, process.exit is correctly called with `1`, but also in the test `src/app.js` is never brought into scope (not saying it should be). So, when we start our app with its usual entrypoint, it looks like the handler defined in `src/app.js` is registered before this one. So while this test may pass here, I'm not sure it is representative of what would happen if there really was an `uncaughtException`.

For context, I think the one in`src/app.js` was defined because the many attempts (and failures) to connect to redis in CI caused the tests to fail abruptly. I think this is still needed.